EventFilter/HcalRawToDigi/src/HcalDataFrameFilter.cc

```diff
@@ -92,6 +129,29 @@ ZDCDigiCollection HcalDataFrameFilter::filter(const ZDCDigiCollection& incol, Hc
   return output;
 }
 
+QIE10DigiCollection HcalDataFrameFilter::filter(const QIE10DigiCollection& incol, HcalUnpackerReport& r) {
+  QIE10DigiCollection output(incol.samples());
+  for (QIE10DigiCollection::const_iterator i=incol.begin(); i!=incol.end(); i++) {
+    if (!HcalDataFrameFilter_impl::checkQIE10(*i,requireCapid_,requireDVER_))
+      r.countBadQualityDigi(i->id());
+    // Never exclude QIE10 digis as their absence would be
+    // treated as a digi with zero charged deposited in that channel
+    output.push_back(*i);
```

Comment:
1) There was no filter at all for QIE10 or QIE11 digis before this PR. The other PRs to ignore invalid hits are due to the (new) presence of calibration channels in the regular stream.
2) Quality checks are done in the `HFPhase1Reconstructor`. In the dual anode reconstruction, the meaning of "no digi" is different from the meaning of "bad digi". @igv4321, please confirm/elaborate if I missed something.